gpu/src/metal/abstractions/state.rs

```diff
@@ -74,6 +74,16 @@ impl MetalState {
         )
     }
 
+    pub fn set_bytes<T>(index: usize, data: &[T], encoder: &ComputeCommandEncoderRef) {
```

Comment:
That seems like a good idea but in reality the side effects are in the command buffer, not in the encoder. The encoder has a ref to the buffer and writes to it. Also side effects are somewhat reflected in the fact that the encoder exists, because when I want to start writing commands I create it and when I finish I need to call `end_encoding()`, wdyt?